readme.md

```diff
@@ -28,3 +28,74 @@ docker-compose up
 # Run the E2E tests:
 
 `./run_tests.py`
+
+# Database Stuff
+
+## What is a migration?
+
+Prior to the fork of themotte from rDrama, there were no database migrations, and the database schema was stored in `schema.sql`. Any time a column or such was added to a model, one hoped that the author remembered to update `schema.sql` to add that column. One of the first changes we made after forking this repo was to add database migrations using Alembic.
+
+Database migrations are instructions for how to convert an out-of-date database into an up-to-date database. This can involve changing the schema, or changing data within the database.
+
+## Why use database migrations
+
+Database migrations allow us to specify where data moves when there are schema changes. This is important when we're live -- if we rename the `comments.ban\_reason` column to `comments.reason\_banned` for naming consistency or whatever, and we do this by dropping the `ban\_reason` column and adding a `reason\_banned` column, we will lose all user data in that column. We don't want to do this. With migrations, we could instead specify that the operation in question should be a column rename, or, if the database engine does not support renaming columns, that we should do a three-step process of "add new column, migrate data over, drop old column".
+
+## Database schema change workflow
+
+As an example, let's say we want to add a column `is\_filtered` to the `comments` table.
```

Comment:
This line says `is_filtered`, but the actual code is `is_flagged`. I *think* this is the wrong one, but either way, something needs to be fixed :D

Reply:
```suggestion
As an example, let's say we want to add a column `is\_flagged` to the `comments` table.
```